crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs

```diff
@@ -75,24 +76,266 @@ impl CreateCommand for TestParams {
         self.origin_id.clone()
     }
 
-    fn create_unit_graph_command(&self, root: PathBuf) -> Command {
-        cargo_command_with_unit_graph("test", root)
+    fn get_targets_ids(&self) -> Vec<BuildTargetIdentifier> {
+        self.targets.clone()
+    }
+
+    fn create_unit_graph_command(&self, root: &Path, targets_details: &[TargetDetails]) -> Command {
+        let targets_args = targets_details_to_args(targets_details);
+        cargo_command_with_unit_graph(CommandType::Test, root, targets_args)
     }
 
-    fn create_requested_command(&self, root: PathBuf) -> Command {
-        // TODO add appropriate build target to arguments
-        let mut cmd = Command::new(toolchain::cargo());
-        cmd.current_dir(root)
-            .args([
-                "test",
-                "--message-format=json",
-                "--",
-                "--show-output",
-                "-Z",
-                "unstable-options",
-                "--format=json",
-            ])
+    fn create_requested_command(&self, root: &Path, targets_details: &[TargetDetails]) -> Command {
+        let targets_args = targets_details_to_args(targets_details);
+        let mut cmd = create_requested_command(CommandType::Test, root, targets_args);
+        cmd.args(["--show-output", "-Z", "unstable-options", "--format=json"])
             .args(self.arguments.clone());
         cmd
     }
 }
+
+impl TargetDetails {
+    pub fn get_enabled_features_str(&self) -> Option<String> {
```

Comment:
jaka jest różnica między pustym stringiem a None?

Reply:
To jeszcze z Brancha Patryka było, nie jako część moich zmian